### ISSUE TITLE
I think this issue has the same root cause as #54245 and #52663 and it was actually not fixed by switching from thrust to cub. The cause should be the `STB_GNU_UNIQUE` symbols as pointed out by @VoVAllen in https://github.com/NVIDIA/thrust/issues/1401#issuecomment-806403746.

### ISSUE THREAD

**Issue by sonamRinchen** [pytorch/pytorch#60994]
Hi, I ran into similar issue as well. I tried almost all the fixes that you guys have mentioned above but sadly i still could not able to fix the problem. I even tried the fix that @ysyyork and @ppwwyyxx have suggested to use that python script. But I still have the same issue though. Please help solve this issue. I really appreciate the help. Thank you so much

my system configurations: python3.7, pytorch 1.9.0+cu111, torchvision 0.10.0+cu111 and Cuda 11.1

**Comment by sonamRinchen** [comment]
![Screenshot from 2021-06-29 15-34-52](https://user-images.githubusercontent.com/53063546/123873106-fac0ca00-d903-11eb-923a-161071a5b7e0.png)

This is how my error message looks like.

**Comment by mruberry** [comment]
cc @malfet @ngimel

**Comment by sonamRinchen** [comment]
Can you please let me know if there is any bug (cuda device-side error) free version of pytorch for cuda 11.x?  @ngimel

**Comment by ngimel** [comment]
Please provide a self-contained script reproducing your error.

**Comment by malfet** [comment]
Also, please run `python3 -m torch.utils.collect_env` and share its output as recommended in bug reporting template.

**Comment by sonamRinchen** [comment]
![Screenshot from 2021-06-29 20-11-13](https://user-images.githubusercontent.com/53063546/123883086-32387200-d916-11eb-9bae-e49ababc7682.png)

Thank you @malfet  and Here is screenshot of my env

**Comment by malfet** [comment]
If you have a 1080 GPU, why do you need 1.9.0-cu111 version of PyTorch? CUDA-10.2 build does not suffer from this issue (as torch_cuda library is not split)

**Comment by sonamRinchen** [comment]
> If you have a 1080 GPU, why do you need 1.9.0-cu111 version of PyTorch? CUDA-10.2 build does not suffer from this issue (as torch_cuda library is not split) and have the same performance on 1080 gpu as 11.1

To be honest i tried installing CUDA 10.2 but my nvidia-driver version is 460 which is not compatible. This  nvidia driver version could only be compatible with  CUDA 11.x. Also, I can not change nvidia-driver because my other software systems needed this latest version of nvidia-driver.

**Comment by sonamRinchen** [comment]
Hi @malfet, thank you for the reply. I will try to install everything again in a different machine. Could you please let me know the best version of pytorch and torchvision for CUDA 10.2 which would not gonna give me this error? Thank u so much again for your help

**Comment by ngimel** [comment]
You error is most likely caused by invalid user data you are sending, and won't be fixed by changing cuda versions.

**Comment by malfet** [comment]
> To be honest i tried installing CUDA 10.2 but my nvidia-driver version is 460 which is not compatible. This  nvidia driver version could only be compatible with  CUDA 11.x. Also, I can not change nvidia-driver because my other software systems needed this latest version of nvidia-driver.

Not sure I understand what do you mean by "driver version is 460 which is not compatible". CUDA gives very strong backward compatibility guarantee, so cuda-11 driver is definitely compatible with cuda-10 runtime. 

What error are you seeing when trying to install pytorch build again cuda 10.2 on your machine?

**Comment by sonamRinchen** [comment]
> You error is most likely caused by invalid user data you are sending, and won't be fixed by changing cuda versions.

Thank You @ngimel for your reply. I kinda check every little thing about data and i even printed how my inputs and label looks like. It could not find any problem there but you never know. Following is my code, please let me know if i am doing anything wrong here.  Thank you so much again for you help.

import os
import numpy as np
import torch
from PIL import Image
from engine import train_one_epoch, evaluate
import utils       
import transforms as T
import torchvision
import json
import cv2
from torchvision.models.detection.faster_rcnn import FastRCNNPredictor
from torchvision.models.detection.mask_rcnn import MaskRCNNPredictor
from torchvision.models.detection import FasterRCNN
from torchvision.models.detection.rpn import AnchorGenerator

class PennFudanDataset(object):

     def __init__(self, root, transforms):
       
        self.root = root
        self.transforms = transforms
        self.imgs = list(sorted(os.listdir(os.path.join(root, "img"))))
        self.json_labels = self.get_json()
        #self.masks = list(sorted(os.listdir(os.path.join(root, "PedMasks"))))
        
     def get_json(self):
        json_file_path = os.path.join(self.root,"det_label.json")
        with open(json_file_path) as f:
            json_labels = json.load(f)
        return json_labels
        
     def __getitem__(self, idx):
        categories = ["pedestrian", "rider", "car", "truck", "bus", "train", "motorcycle", "bicycle", "traffic light", "traffic sign"]
        cat_ids, boxes = [],[]
        target = {}
        # load images and masks
        img_path = os.path.join(self.root, "img", self.imgs[idx])
        json_label = self.json_labels[idx]        
        img = cv2.imread(img_path)    #Image.open(img_path).convert("RGB")
     
        for l in json_label["labels"]:            
            for id, cat in enumerate(categories, start=1):
                if cat == l["category"]:
                    cat_ids.append(id)
                    #print("class", l["category"])
                    #print("class", l["box2d"])  
            box = l["box2d"]
            boxes.append([box["x1"], box["y1"], box["x2"],  box["y2"]])
            #print("Category", l["category"])  
        target["labels"] = torch.as_tensor(cat_ids, dtype=torch.int64)
        target["boxes"] = torch.as_tensor(boxes, dtype=torch.float32)
        target["image_id"] = torch.tensor([idx])
        if self.transforms is not None:
            img, target = self.transforms(img, target)
            
        print("img name, label name",self.imgs[idx],json_label["name"])
        print("catID:", target['labels'])
        return img, target

     def __len__(self):s
        return len(self.imgs)
        
def get_model_instance_segmentation(num_classes):
    # load an instance segmentation model pre-trained pre-trained on COCO
    model = torchvision.models.detection.fasterrcnn_resnet50_fpn(pretrained=True)
    
    in_features = model.roi_heads.box_predictor.cls_score.in_features
    
    model.roi_heads.box_predictor = FastRCNNPredictor(in_features, num_classes)
  
    return model  
         
def get_transform(train):
    transforms = []
    transforms.append(T.ToTensor())
    if train:
        transforms.append(T.RandomHorizontalFlip(0.5))
    return T.Compose(transforms)     
     
        
def train():
    # train on the GPU or on the CPU, if a GPU is not available
    
    os.environ['CUDA_LAUNCH_BLOCKING']= "1"
    device = torch.device('cuda') if torch.cuda.is_available() else torch.device('cpu')
    #device = torch.device('cpu')
    num_classes = 11
    # use our dataset and defined transformations
    dataset = PennFudanDataset(os.path.join('det_final_label_image','train'), get_transform(train= True))
    dataset_test = PennFudanDataset(os.path.join('det_final_label_image','test'), get_transform(train=False))

    # split the dataset in train and test set
    torch.manual_seed(1)
    indices = torch.randperm(len(dataset)).tolist()
    dataset = torch.utils.data.Subset(dataset, indices[:-50])
    dataset_test = torch.utils.data.Subset(dataset_test, indices[-50:])

    # define training and validation data loaders
    data_loader = torch.utils.data.DataLoader(
    dataset, batch_size=2, shuffle=True, num_workers=7,
    collate_fn=utils.collate_fn)

    data_loader_test = torch.utils.data.DataLoader(
    dataset_test, batch_size=1, shuffle=False, num_workers=7,
    collate_fn=utils.collate_fn)

    # get the model using our helper function
    model = get_model_instance_segmentation(num_classes)

    # move model to the right device
    model.to(device)

    # construct an optimizer
    params = [p for p in model.parameters() if p.requires_grad]
    optimizer = torch.optim.SGD(params, lr=0.005,
                                momentum=0.9, weight_decay=0.0005) 
                               
    # and a learning rate scheduler
    lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer,
                                                   step_size=1,
                                                   gamma=0.1)

    # let's train it for 10 epochs
    num_epochs = 10

    for epoch in range(num_epochs):
    
        train_one_epoch(model, optimizer, data_loader, device, epoch, print_freq=10)
        # update the learning rate
        lr_scheduler.step() 
        # evaluate on the test dataset
        evaluate(model, data_loader_test, device=device)

    print("That's it!")
        
if __name__ == "__main__":
    #learn.destroy()
    train()

**Comment by sonamRinchen** [comment]
> > To be honest i tried installing CUDA 10.2 but my nvidia-driver version is 460 which is not compatible. This  nvidia driver version could only be compatible with  CUDA 11.x. Also, I can not change nvidia-driver because my other software systems needed this latest version of nvidia-driver.
> 
> Not sure I understand what do you mean by "driver version is 460 which is not compatible". CUDA gives very strong backward compatibility guarantee, so cuda-11 driver is definitely compatible with cuda-10 runtime.
> 
> What error are you seeing when trying to install pytorch build again cuda 10.2 on your machine?

@malfet thank you again for your reply. Well as far the CUDA 10.2 my nvidia driver (460) doesnot supports it. But I can get another machine with CUDA 10.2. Can you please let me know which version of pytorch and torchvision would not cause this "CUDA device side error triggered"? I will build everything again in a different machine. Thank you so much again.

**Comment by ngimel** [comment]
As @malfet said, 460 driver is compatible with all older cuda runtime versions, so you definitely should be able to run cuda 10.2 pytorch on this machine. 
The script you provided is not self-contained reproducing script, it has external data dependencies, thus it's not actionable for us.

**Comment by sonamRinchen** [comment]
> As @malfet said, 460 driver is compatible with all older cuda runtime versions, so you definitely should be able to run cuda 10.2 pytorch on this machine.
> The script you provided is not self-contained reproducing script, it has external data dependencies, thus it's not actionable for us.

Thank you @ngimel for your help. I will try to install CUDA 10.2 again. Also, what version of pytorch and torch vision i should install with CUDA10.2? 

I was using BDD100K dataset (for the labels i was Det_20 of BDD dataset). Either way I understand it would be hard to reproduce this error at your end. Thank you so much you time and help.
